src/hook/hook.cpp

```diff
@@ -772,6 +774,8 @@ int WINAPI NWNXWinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance,
 
 	shmem = nullptr;
 
+	std::this_thread::sleep_for(10s);

```

Comment:
This seems potentially questionable; what are we delaying for?

Reply:
This is all a WIP. I placed it in there so I could space out my debugging of the shared memory object. I found the issue, so I've removed it locally and will be removed as I start cleaning the code.

Reply:
I see.

Useful debugging techniques for this sort of situation that I usually use, for reference:

1) Putting in an if (IsDebuggerPresent())) __debugbreak();
2) windbg -o app (or .childbg 1) so that child processes are debugged too.